hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/testutils/SparkClientFunctionalTestHarness.java

```diff
@@ -228,8 +228,8 @@ protected void insertRecords(HoodieTableMetaClient metaClient, List<HoodieRecord
 
     roView = getHoodieTableFileSystemView(reloadedMetaClient, hoodieTable.getCompletedCommitsTimeline(), allFiles);
     dataFilesToRead = roView.getLatestBaseFiles();
-    assertTrue(dataFilesToRead.findAny().isPresent(),
-        "should list the base files we wrote in the delta commit");
+    assertEquals(!hoodieTable.getIndex().canIndexLogFiles(), dataFilesToRead.findAny().isPresent(),
```

Comment:
is this change related? I dont fully get the reasoning. I think base files can still be present (for example, after compaction/clustering) even if canIndexLogFiles is true

Reply:
This method is being used to do first couple of inserts before the compaction/clustering is triggered. With canIndexLogFiles true, those inserts will only write log files. However, I see your point. I think we should not have this assertion here at all. It should be within respective tests. This method should only be concerned with inserting records. What do you think?

Reply:
yes, +1 to revert this change and keep test logic within respective suites